src/main/java/org/reactome/release/cosmicupdate/COSMICIdentifierUpdater.java

```diff
@@ -0,0 +1,194 @@
+	}
+	public boolean isValid()
+	{
+		return valid;
+	}
+	public void setValid(boolean valid)
+	{
+		this.valid = valid;
+	}
+	public Set<String> getMutationIDs()
+	{
+		return mutationIDs;
+	}
+	public void setMutationIDs(Set<String> mutationIDs)
+	{
+		this.mutationIDs = mutationIDs;
+	}
+	public String getCosvIdentifier()
+	{
+		return cosvIdentifier;
+	}
+	public void setCosvIdentifier(String cosvIdentifier)
+	{
+		this.cosvIdentifier = cosvIdentifier;
+	}
+	
+	@Override
+	public String toString()
+	{
+		return "[" + this.getDbID() + "; " 
+					+ this.getIdentifier()+ "; "
+					+ this.getSuggestedPrefix() + "; "
+					+ this.isValid() + "; "
+					+ this.getCosvIdentifier() + "; ("
+					+ this.getMutationIDs().stream().reduce("", (s, t) -> s + t + ",") + ")" + "]";
+	}
+	
+	/**
+	 * Implementation of compareTo ensures that these are sorted by validity, and then by identifier.
+	 * This is done so that the invalid identifiers are at the top of the report.
+	 */
+	@Override
+	public int compareTo(COSMICIdentifierUpdater other)
+	{
+		if (other.valid && !this.valid)
+		{
+			return -1;
+		}
+		if (!other.valid && this.valid)
+		{
+			return 1;
+		}
+		if (other.valid == this.valid)
+		{
+			return this.identifier.compareTo(other.identifier);
+		}
+		return 0;
+	}
+
+	/**
+	 * Perform an update of a COSMIC identifier.
+	 * @param adaptor - the database adapter to use.
+	 * @param creatorID - the DB_ID of the Creator of this update.
+	 * @throws Exception
+	 */
+	public void updateIdentfier(MySQLAdaptor adaptor, long creatorID) throws Exception
+	{
+		// If there is a COSV identifier, we'll update using that.
+		if (this.getCosvIdentifier() != null && !this.getCosvIdentifier().isEmpty())
+		{
+			// Create the InstanceEdit, if necessary.
+			synchronized (COSMICIdentifierUpdater.class)
+			{
+				if (COSMICIdentifierUpdater.updateWithNewCOSV == null)
+				{
+					COSMICIdentifierUpdater.updateWithNewCOSV = InstanceEditUtils.createDefaultIE(adaptor, creatorID, true, "Identifier was automatically updated to new \"COSV\" identifier by COSMIC Update process.");
+				}
+			}
+			GKInstance identifierObject = adaptor.fetchInstance( this.getDbID());
+			
+			updateIdentifierObject(adaptor, COSMICIdentifierUpdater.updateWithNewCOSV, identifierObject, this.getCosvIdentifier());
+		}
+		// If no COSV identifier was found, update using the suggested prefix (determined computationally).
+		else if (this.getSuggestedPrefix() != null && this.getSuggestedPrefix().equalsIgnoreCase(COSMICUpdateUtil.COSMIC_LEGACY_PREFIX))
+		{
+			GKInstance identifierObject = adaptor.fetchInstance( this.getDbID());
+			String currentIdentifier = (String) identifierObject.getAttributeValue(ReactomeJavaConstants.identifier);
+			// If the current identifier already begins with "C" then leave it alone.
+			// This code is for updating numeric identifiers that have a suggested prefix.
+			if (!currentIdentifier.toUpperCase().startsWith("C"))
```

Comment:
Could this check be for if the currentIdentifier starts with an integer? That way it stays in sync with the commenting, and in case in the future if they introduce non-C-starting identifiers (introducing KOSMIC!), it can handle that

Reply:
Checking that the first character is numeric would require a regex (or parsing a string to numeric) vs. checking that the first character is "C" is simpler.

Reply:
I understand the point about regexes, but I think checking for a number would be better since the code will attempt to prepend "COSM" anything that doesn't start with a C.  This isn't a problem now, but it is vulnerable to wrongly pretending "COSM" if a different identifier exists later.

Checking the precise case of if it is an integer seems safer despite a slightly more expensive check so we know it only works with that kind of identifier.

Reply:
COSMIC seems pretty committed to their "COS-" notation. I think we can change the code to handle other identifiers once COSMIC starts using a different notation system.